thoth/prescriptions_refresh/handlers/image_analysis.py

```diff
@@ -0,0 +1,218 @@
+
+    if response.status_code == 200:
+        results = response.json()
+        return results[0]
+    else:
+        return None
+
+
+def _get_requirement_files_from_image_analysis(
+    package_extract_document_id: str,
+) -> Tuple[Optional[Dict[str, Any]], Optional[Dict[str, Any]]]:
+    """Get requirements files from image analysis result."""
+    url = f"http://{USER_API_HOST}/api/v1/analyze"
+    response = requests.get(url, params={"analysis_id": package_extract_document_id})
+
+    if response.status_code == 200:
+        document = response.json()
+
+        # Get result and requirements files
+        result = document["result"]
+        pipfile_dict = result["aicoe-ci"].get("requirements")
+        pipfile_lock_dict = result["aicoe-ci"].get("requirements_lock")
+
+        return pipfile_dict, pipfile_lock_dict
+
+    elif response.status_code == 404:
+        _LOGGER.warning(f"Document {package_extract_document_id} does not exist.")
+        return None, None
+
+    else:
+        return None, None
+
+
+def _create_resolved_dependencies_section(pipfile_d: Dict[str, Any], pipfile_lock_d: Dict[str, Any]) -> str:
+    """Create resolved dependencies section for adviser unit."""
+    resolved_dependencies = ""
+
+    pipfile = Pipfile.from_dict(pipfile_d)
+    pipfile_lock = PipfileLock.from_dict(pipfile_lock_d)
+
+    required_packages = []
+
+    for package in pipfile.packages.packages:
+        if package in _REPRESENTATIVE_PACKAGES:
+            version = pipfile_lock.packages[package].version
+            required_packages.append({"name": package, "version": version})  # Includes ==
+
+    line = 0
+    for package in required_packages:
+        if line == 0:
+            resolved_dependencies += f"- name: {package['name']}\n"
+        else:
+            resolved_dependencies += f"        - name: {package['name']}\n"
+
+        resolved_dependencies += f"          version: {package['version']}\n"
+        line += 1
+
+    return resolved_dependencies
+
+
+def thoth_image_analysis(prescriptions: "Prescriptions") -> None:
+    """Create prescriptions from Thoth container image analysis results."""
+    if not QUAY_TOKEN:
+        raise ValueError("No Token to Quay API provided")
+
+    for image in sorted(chain(get_ps_s2i_image_names(), get_configured_image_names())):
+
+        # Get tag for Thoth images hosted on Quay
+        for _, tag in get_image_containers(image):
+            _LOGGER.info("Obtaining image information for %r in tag %r", image, tag)
+
+            image_url = f"{QUAY_URL}/thoth-station/{image}:{tag}"
+
+            # Get image analyzed IDs latest from database through USER-API endpoint
+            info = _get_latest_image_analyzed_info(image=image)
+
+            if not info:
+                _LOGGER.warning("Could not find any data for {QUAY_URL}/thoth-station/{image}:{tag} in Thoth Database.")
+                break
```

Comment:
Other tags could be checked as well (or I missed the flow here):

```suggestion
                continue
```

(here and bellow)

Reply:
correct, thanks!

Reply:
Does this suggestion still apply?